models/Events/DeliveryExecutionResultsRecalculated.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace oat\taoResultServer\models\Events;
+
+use oat\oatbox\event\Event;
+use oat\taoDelivery\model\execution\DeliveryExecutionInterface;
+
+class DeliveryExecutionResultsRecalculated implements Event
+{
+    private DeliveryExecutionInterface $deliveryExecution;
+    private ?float $totalScore;
```

Comment:
you cast it before passing here so "?" is pointless, it will be 0.0 always

Reply:
Yess sorry for confusion I was refering to this part of code 
` new DeliveryExecutionResultsRecalculated($deliveryExecution, (float)$scoreTotal, (float)$scoreTotalMax)`
So input value is always cast to float, so no null here.